functors_task_1/student/include/impl.h

```diff
@@ -40,9 +48,20 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink bev1, NonAlcoholDrink bev2)
     {
-    }
+		switch (bev1)
+		{
+		case AlcoholDrink::Gin:
+			if (bev2 == NonAlcoholDrink::LimeJuice) return Cocktail::Gimlet;
```

Comment:
This can make code more readable:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice